### PR TITLE
Application Import Export Support for APIM 2.2.0

### PR DESCRIPTION
**Pull request by randilu**
### Proposed changes in this pull request

- This is the Application Import Export Support for WSO2 API Manager 2.2.0
- This enables users to migrate applications between different environments with or without adding subscriptions as desired by the importing user
- Users are able to export an application as a zip archive file, which in return can be imported to a 
 another environment (dev/QA)
- This also supports import/exports in Multi-Tenanted environments as well.

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

- Docs : https://docs.google.com/document/d/1i_tyr-GlloVujhxxADeQTqtBu3rOFHq8aLZJwdkWJ4U

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ExportApiService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package org.wso2.carbon.apimgt.rest.api.admin;
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ExportApiService.java]: Please add license headers.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApiService.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.wso2.carbon.apimgt.rest.api.admin;
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApiService.java]: Please add license header here also. Check other classes as well

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
+package org.wso2.carbon.apimgt.rest.api.admin.dto;
+
+
+import io.swagger.annotations.*;
+import com.fasterxml.jackson.annotation.*;
+
+import javax.validation.constraints.NotNull;
+
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: Remove unnecessary spaces.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+
+
+
+@ApiModel(description = "")
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: Add valid description here.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import org.wso2.carbon.apimgt.rest.api.admin.dto.APIInfoDTO;
+
+import io.swagger.annotations.*;
+import com.fasterxml.jackson.annotation.*;
+
+import javax.validation.constraints.NotNull;
+
+
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: remove spaces here as well

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        Response.ResponseBuilder responseBuilder = Response.status(Response.Status.OK)
+                .entity(exportedApplicationArchiveFile).type(MediaType.APPLICATION_OCTET_STREAM);
+        responseBuilder
+                .header("Content-Disposition", "attachment; filename=\"" + exportedFileName + "\"");
+        return responseBuilder.build();
+    }
+
+    @Override
+    public String exportApplicationsGetGetLastUpdatedTime(String appName, String appOwner) {
+        return null;
```
Comment (Krishanx92) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java]: why this method return null?

Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java]: This generated method cannot be implemented

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ExportApiService.java (hunk 1 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ExportApiService.java]: Add licence header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApi.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
+package org.wso2.carbon.apimgt.rest.api.admin;
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApi.java]: Add license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApiService.java (hunk 2 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApiService.java]: Add license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
+package org.wso2.carbon.apimgt.rest.api.admin.dto;
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: Add license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
+package org.wso2.carbon.apimgt.rest.api.admin.dto;
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: Need license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ExportApiServiceFactory.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package org.wso2.carbon.apimgt.rest.api.admin.factories;
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ExportApiServiceFactory.java]: add license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ImportApiServiceFactory.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
+package org.wso2.carbon.apimgt.rest.api.admin.factories;
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ImportApiServiceFactory.java]: Add license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+  private String lastUpdatedTime = null;
+
+  private String createdTime = null;
+
+  /**
+  * gets and sets the lastUpdatedTime for APIInfoDTO
+  **/
+  @JsonIgnore
+  public String getLastUpdatedTime(){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    return lastUpdatedTime;
+  }
+  public void setLastUpdatedTime(String lastUpdatedTime){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+    this.lastUpdatedTime=lastUpdatedTime;
+  }
+
+  /**
+  * gets and sets the createdTime for a APIInfoDTO
+  **/
+
+  @JsonIgnore
+  public String getCreatedTime(){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+    return createdTime;
+  }
+  public void setCreatedTime(String createdTime){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+  private String lastUpdatedTime = null;
+
+  private String createdTime = null;
+
+  /**
+  * gets and sets the lastUpdatedTime for APIInfoListDTO
+  **/
+  @JsonIgnore
+  public String getLastUpdatedTime(){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+    return lastUpdatedTime;
+  }
+  public void setLastUpdatedTime(String lastUpdatedTime){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+    this.lastUpdatedTime=lastUpdatedTime;
+  }
+
+  /**
+  * gets and sets the createdTime for a APIInfoListDTO
+  **/
+
+  @JsonIgnore
+  public String getCreatedTime(){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+    return createdTime;
+  }
+  public void setCreatedTime(String createdTime){
```
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApi.java (hunk 7 above)
Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApi.java]: These are generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java (hunk 14 above)
Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: Formatting issues needs to be fixed on the code generator side

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+     * @param appName Search query
+     * @return Zip file containing exported Application
+     */
+    @Override
+    public Response exportApplicationsGet(String appName, String appOwner) {
+        APIConsumer consumer;
+        String exportedFilePath;
+        Application applicationDetails = null;
+        File exportedApplicationArchiveFile = null;
+        String exportedAppDirName = "exported-application";
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java]: Shall we move this String to a constant?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+        String pathToExportDir = System.getProperty(RestApiConstants.JAVA_IO_TMPDIR) + File.separator +
+                "exported-app-archives-" +
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java]: Let's move this to a String constant

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+     * @return imported Application
+     */
+    @Override
+    public Response importApplicationsPost(InputStream fileInputStream, Attachment fileDetail,
+                                           Boolean preserveOwner, Boolean skipSubscriptions, String appOwner) {
+        APIConsumer consumer;
+        String ownerId;
+        String username = RestApiUtil.getLoggedInUsername();
+        String tempDirPath = System.getProperty(RestApiConstants.JAVA_IO_TMPDIR) + File.separator +
+                "imported-app-archive-" +
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java]: Same here

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java [hunk 23]
```diff
@@ -0,0 +1,123 @@
...
+    /**
+     * Import an Application which has been exported to a zip file
+     *
+     * @param fileInputStream content stream of the zip file which contains exported Application
+     * @param fileDetail      meta information of the zip file
+     * @return imported Application
+     */
+    @Override
+    public Response importApplicationsPost(InputStream fileInputStream, Attachment fileDetail,
+                                           Boolean preserveOwner, Boolean skipSubscriptions, String appOwner) {
```
Comment (RenukaFernando) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java]: Shall we update java doc comments

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApiService.java (hunk 2 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/ImportApiService.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java (hunk 8 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java (hunk 9 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ExportApiServiceFactory.java (hunk 10 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ExportApiServiceFactory.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ImportApiServiceFactory.java (hunk 11 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/factories/ImportApiServiceFactory.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java (hunk 12 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java (hunk 13 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java (hunk 15 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java (hunk 16 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java (hunk 17 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java (hunk 18 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java (hunk 19 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/APIInfoListDTO.java]: This is generated code

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java (hunk 23 above)
Comment (randilu) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ImportApiServiceImpl.java]: Done

**Pull request merged**